### PR TITLE
Split package removal

### PR DESCRIPTION
**Pull request by vietj**
This removes split packages of the code generator and make vertx-codegen-api a modular jar.

This is a breaking change for code generator implementations since the generator classes are repacked as `io.vertx.codegen.processor`, generators should be updated accordingly.

This does not add JPMS support to the code generator itself, while this is not precluded, this is a non goal.

### CODE REVIEW

**Comment by vietj** [comment]
cc @jponge

File: pom.xml [hunk 1]
```diff
@@ -33,6 +33,7 @@
     <stack.version>5.0.0-SNAPSHOT</stack.version>
     <doc.skip>true</doc.skip>
     <jar.manifest>${project.basedir}/src/main/resources/META-INF/MANIFEST.MF</jar.manifest>
+    <maven.compiler.release>11</maven.compiler.release>
```
Comment (jponge) [inline pom.xml:36]: Shouldn't we consider bumping to a 17 baseline?

File: pom.xml [hunk 2]
```diff
@@ -55,6 +56,17 @@
   <dependencies>
   </dependencies>
 
+  <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.13.0</version>
```
Comment (jponge) [inline pom.xml:64]: Extract a property instead 😉

**Review by jponge — COMMENTED** [review]
👍

File: pom.xml (hunk 1 above)
Comment (vietj) [reply pom.xml:36]: we stick to 11 for Vert.x 5.0

File: pom.xml (hunk 2 above)
Comment (vietj) [reply pom.xml:64]: actually I will need to properly update this to a new vertx parent, this is still a draft PR kind off, as I want to prepare the vertx-rx / vertx-kotlin for code migration

**Comment by vietj** [comment]
@jponge I removed the JPMS part which relates to the build as I first want to repackage the processor and update the stack accordingly, therefore the commit that introduces the module-info and modifies the build is removed, only the io.vertx.codegen.* classes are repackaged to io.vertx.codegen.processor.*

**Pull request merged**